### PR TITLE
Fix Resource Class reflection registration when custom Writer is used

### PR DESCRIPTION
**Pull request by geoand**
Originally reported [here](https://quarkusio.zulipchat.com/#narrow/stream/306418-resteasy-reactive/topic/CloudEvents.20lib.20doesn't.20work.20in.20native.20mode/near/357052399)

### CODE REVIEW

File: extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/ClientJacksonMessageBodyWriter.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
 import jakarta.ws.rs.WebApplicationException;
 import jakarta.ws.rs.core.MediaType;
 import jakarta.ws.rs.core.MultivaluedMap;
-import jakarta.ws.rs.ext.MessageBodyWriter;
+import `jakarta.ws.rs.ext.MessageBodyWriter`;
```
Comment (Sgitario) [inline extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/ClientJacksonMessageBodyWriter.java:18]: What is the motivation for this change?

**Review by Sgitario — APPROVED** [review]
lgtm

File: extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/ClientJacksonMessageBodyWriter.java (hunk 1 above)
Comment (cescoffier) [reply extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/ClientJacksonMessageBodyWriter.java:18]: I think it won't compile...

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/ClientJacksonMessageBodyWriter.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/ClientJacksonMessageBodyWriter.java]: I have no idea how that happened. It should be fixed now

**Review by cescoffier — APPROVED** [review]
I'm wondering it we should add tests verifying when the "written" type is used in a Uni or Multi, such as:

```java
    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Uni<Message3> out3() {
        return Uni.createFrom().item(Message3("test"));
    }

    @GET
    @Produces(MediaType.APPLICATION_JSON) // Should not matter
    public Multi<Message4> out4() {
        return Multi.createFrom().item(Message4("test"));
    }

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Flow.Publisher<Message5> out5() { 
        return Miulti.createFrom().item(Message5("test"));
    }
```

**Comment by geoand** [comment]
We already have `Multi` tests in another class - that is actually the reason we didn't find this issue in the first place :)
